client/rest/src/main/java/org/elasticsearch/client/Request.java

```diff
@@ -125,21 +126,18 @@ public HttpEntity getEntity() {
     }
 
     /**
-     * Set the headers to attach to the request.
+     * Add the provided header to the request.
      */
-    public void setHeaders(Header... headers) {
-        Objects.requireNonNull(headers, "headers cannot be null");
-        for (Header header : headers) {
-            Objects.requireNonNull(header, "header cannot be null");
-        }
-        this.headers = headers;
+    public void addHeader(Header header) {
```

Comment:
What about `addHeader(String key, String value)` instead? I don't see us using anything but `BasicHeader` anyway and I like exposing a little less of httpclient publicly. And I like not having to import another class everywhere.

Reply:
yea I think it's a good idea